chains/chains.go

```diff
@@ -121,6 +121,53 @@ func Predict(ctx context.Context, c Chain, inputValues map[string]any, options .
 	return outputValue, nil
 }
 
+const _defaultApplyMaxNumberWorkers = 5
+
+// Apply executes the chain for each of the inputs asynchronously.
+func Apply(ctx context.Context, c Chain, inputValues []map[string]any, maxWorkers int, options ...ChainCallOption) ([]map[string]any, error) { // nolint:lll
+	if maxWorkers <= 0 {
+		maxWorkers = _defaultApplyMaxNumberWorkers
+	}
+
+	inputJobs := make(chan map[string]any, len(inputValues))
+	resultsChan := make(chan struct {
+		result map[string]any
+		err    error
+	}, len(inputValues))
+	defer close(inputJobs)
```

Comment:
Nitpick: we can close the `inputJobs` channel immediately after writing inputs to it so we can release the consuming goroutines before the `Apply` method finishes.